launcher/container_runner.go

```diff
@@ -552,13 +560,34 @@ func (r *ContainerRunner) Run(ctx context.Context) error {
 		return fmt.Errorf("failed to measure CEL events: %v", err)
 	}
 
-	if err := r.fetchAndWriteToken(ctx); err != nil {
-		return fmt.Errorf("failed to fetch and write OIDC token: %v", err)
+	// Only refresh token if agent has a default GCA client (not ITA use case).
+	if r.launchSpec.ITARegion == "" {
+		if err := r.fetchAndWriteToken(ctx); err != nil {
+			return fmt.Errorf("failed to fetch and write OIDC token: %v", err)
+		}
 	}
 
 	// create and start the TEE server
 	r.logger.Info("EnableOnDemandAttestation is enabled: initializing TEE server.")
-	teeServer, err := teeserver.New(ctx, path.Join(launcherfile.HostTmpPath, teeServerSocket), r.attestAgent, r.logger)
+
+	attestClients := &teeserver.AttestClients{}
+	if r.launchSpec.ITARegion != "" {
+		itaClient, err := ita.NewClient(r.launchSpec.ITARegion, r.launchSpec.ITAKey)
+		if err != nil {
+			return fmt.Errorf("failed to create ITA client: %v", err)
+		}
+
+		attestClients.ITA = itaClient
+	} else {
```

Comment:
Some customers need both to help transition or just because some of their end users want to root in intel and some want to root in Google.

I think we can just remove the if-else and create both

Reply:
I think it'd be better to create each client as-needed. The GCA client in particular requires RPCs (Get/List Location) to be initialized, and we shouldn't introduce those potential failure points unless the customer actually wants a GCA token (meaning they request it ODA). 

If the GCA is having problems - for example - we wouldn't want ITA-only customers to be blocked by GCA client initialization. A way around this could be to fail-open on creating the GCA client in that case, but I think that only further complicates things.